### PR TITLE
chore!: refactor testnode to expose appcreator

### LINKED ISSUE #1990: Refactor testnode to expose appcreator
When creating fraudulent blocks, we will need to be able to build a testnode with a slightly different application. In order to do this for the testnode, we should refactor it so that it exposes the [appcreator function currently used](https://github.com/celestiaorg/celestia-app/blob/b2fc98ccdea77fc00deaf56ddd4bd9e4258ca96d/test/util/testnode/full_node.go#L107)

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

This refactor is mainly targeting #1990, but doing so in a way that uses common patters to avoid continuing to blow up the api

closes #1990

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: test/util/testnode/config.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	// ConsensusParams are the consensus parameters of the test node.
+	ConsensusParams *tmproto.ConsensusParams
+	// AppOptions are the application options of the test node.
+	AppOptions *KVAppOptions
+	// GenesisOptions are the genesis options of the test node.
+	GenesisOptions []GenesisOption
+	// Accounts are the accounts of the test node.
+	Accounts []string
+	// AppCreator is used to create the application for the testnode.
+	AppCreator srvtypes.AppCreator
```
Comment (evan-forbes) [inline test/util/testnode/config.go:36]: per #1990, we can now replace this function with a new one that creates a malicious applicaiton

**Review by cmwaters — APPROVED** [review]
Nice cleanup

File: x/qgb/client/verify.go [hunk 2]
```diff
@@ -189,11 +189,11 @@ func sharesCmd() *cobra.Command {
 func VerifyShares(ctx context.Context, logger tmlog.Logger, config VerifyConfig, height uint64, startShare uint64, endShare uint64) (isCommittedTo bool, err error) {
 	trpc, err := http.New(config.TendermintRPC, "/websocket")
 	if err != nil {
-		return
+		return false, err
```
[the remark this answers is not in the document]
Comment (rootulp) [reply x/qgb/client/verify.go:192]: TBH I'm surprised golangci-lint didn't catch this b/c it's a naked return and we have https://github.com/celestiaorg/celestia-app/blob/b2fc98ccdea77fc00deaf56ddd4bd9e4258ca96d/.golangci.yml#L10

**Review by rootulp — APPROVED** [review]
Looks cleaner!

**Pull request merged**